src/main/java/uk/gov/ons/census/fwmt/csvservice/service/impl/CSVConverterServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.ons.census.fwmt.csvservice.service.impl;
+
+import com.opencsv.bean.CsvToBean;
+import com.opencsv.bean.CsvToBeanBuilder;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Service;
+import uk.gov.ons.census.fwmt.common.error.GatewayException;
+import uk.gov.ons.census.fwmt.csvservice.canonical.CanonicalJobHelper;
+import uk.gov.ons.census.fwmt.csvservice.dto.CSVRecordDTO;
+import uk.gov.ons.census.fwmt.csvservice.service.CSVConverterService;
+import uk.gov.ons.census.fwmt.events.component.GatewayEventManager;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalTime;
+
+import static uk.gov.ons.census.fwmt.csvservice.config.GatewayEventsConfig.CSV_REQUEST_EXTRACTED;
+
+@Slf4j
+@Service
+public class CSVConverterServiceImpl implements CSVConverterService {
+  @Value("${gcpBucket.location}")
+  private Resource path;
+
+  @Autowired
+  private CSVAdapterServiceImpl csvAdapterService;
```

Comment:
Not a criticism, but why do you think we need the csvAdapterService, could we not just copy the 
   jobServiceProducer.sendMessage(ceCreateCase);
and event into this class. 

If you have a good answer, we can just leave it!!

Reply:
Currently its only handling CE but eventually it will also handle CCS as as the service increases in size would be nicer separated a bit maybe?

Reply:
but then again the actual conversion will probably happen in the canonical helper